### PR TITLE
Opt-in to transitive content flow

### PR DESCRIPTION
**Pull request by azeno**
This PR enables the control of the ``Include`` attribute for a ``dependency`` with the ``IncludeAssets`` metadata. See https://github.com/NuGet/Home/wiki/%5BSpec%5D-Managing-dependency-package-assets#nuspec for details.

Consider ProjectA with
``
<ProjectReference Include="ProjectB" IncludeAssets="all" />
``
will result in a ProjectA.nuspec with
``
<dependency id="ProjectB" include="all" />
``
which when ProjectA.nuspec gets referenced will also pull in the content of ProjectB into the root project.

### CODE REVIEW

**Comment by azeno** [comment]
I've added two tests for the previous commit.

File: src/Build/NuGet.Build.Packaging.Tasks/Extensions.cs [hunk 1]
```diff
@@ -34,6 +34,15 @@ public static string GetNullableMetadata(this ITaskItem taskItem, string metadat
 			return value;
 		}
 
+		public static string GetContentFileInclude(this ITaskItem taskItem)
```
Comment (kzu) [inline src/Build/NuGet.Build.Packaging.Tasks/Extensions.cs]: this is mixed in in this PR, it belongs to a different one that likely won't get merged soon. It would be best to separate it.

File: src/Build/NuGet.Build.Packaging.Tests/CreatePackageTests.cs [hunk 2]
```diff
@@ -149,6 +149,56 @@ public void when_creating_package_with_simple_dependency_then_contains_dependenc
...
+			task.Contents = new[]
+			{
+				new TaskItem("Newtonsoft.Json", new Metadata
+				{
+					{ MetadataName.PackageId, task.Manifest.GetMetadata("Id") },
+					{ MetadataName.Kind, PackageItemKind.Dependency },
+					{ MetadataName.Version, "8.0.0" },
+					// NOTE: AssignPackagePath takes care of converting TFM > short name
+					{ MetadataName.TargetFramework, "net45" },
+					{ MetadataName.IncludeAssets, "all" }
```
Comment (kzu) [inline src/Build/NuGet.Build.Packaging.Tests/CreatePackageTests.cs:164]: This should be quite useful indeed. I'd add support for `ExcludeAssets` too, to complete the feature.

**Comment by azeno** [comment]
Indeed - I'll rebase the branch.

File: src/Build/NuGet.Build.Packaging.Tasks/CreatePackage.cs [hunk 3]
```diff
@@ -336,6 +349,10 @@ class Dependency
 			public NuGetFramework TargetFramework { get; set; }
 
 			public VersionRange Version { get; set; }
+
+			public string Include { get; set; }
```
Comment (kzu) [inline src/Build/NuGet.Build.Packaging.Tasks/CreatePackage.cs]: This looks unrelated to the Include/Exclude assets, no?

File: src/Build/NuGet.Build.Packaging.Tasks/CreatePackage.cs [hunk 4]
```diff
@@ -286,6 +289,16 @@ static VersionRange AggregateVersions(VersionRange aggregate, VersionRange next)
 			return versionSpec.ToVersionRange();
 		}
 
+		static List<string> AggregateIncludes(List<string> aggregate, string next)
```
Comment (kzu) [inline src/Build/NuGet.Build.Packaging.Tasks/CreatePackage.cs]: Maybe rename to `AggregateAssetsFlow`? Since it now does Includes and Excludes?

Comment (azeno) [reply src/Build/NuGet.Build.Packaging.Tasks/CreatePackage.cs]: Will do

**Review by kzu — APPROVED** [review]

**Comment by kzu** [comment]
thanks a lot for your contribution @azeno!
